sys/arm64/include/cheri.h

```diff
@@ -37,6 +37,19 @@
 #define	__USER_DDC	((void * __capability)curthread->td_frame->tf_ddc)
 #define	__USER_PCC	((void * __capability)curthread->td_frame->tf_elr)
 
+/*
+ * Special marker NOPs for the Morello FVP to start / stop region of interest
+ * in trace.
+ */
+#define	CHERI_START_TRACE	do {					\
+	__asm__ __volatile__("hlt 0xbeef");				\
+} while(0)
+#define	CHERI_STOP_TRACE	do {					\
+	__asm__ __volatile__("hlt 0xbeef");				\
+} while(0)
+
+struct thread;
```

Comment:
Adding this struct thread decl in ebc01143295a35c4332f00325aaae1e55c55bffd along with the tracing macros seems out of place.

Reply:
Yeah that is a bit gross, I added it because promoting `ASM_PTR_CONSTR` means that I need to include the file from atomic.h and need to forward-declare struct thread to avoid cyclic include. The fact that it is in this commit is a mismerge from a rebase and will move it. Do you have a better place for the `ASM_PTR_CONTSR` macros in mind?